src/DotVVM.Framework/Compilation/Binding/ExpressionHelper.cs

```diff
@@ -252,6 +259,53 @@ private static MethodRecognitionResult TryCallMethod(MethodInfo method, Type[] t
             };
         }
 
+        private static Type ResolveGenericTypeFromGivenExpressions(int i, Type genericArgument, ParameterInfo[] parameters, Expression[] args)
+        {
+            for (var j = 0; j < parameters.Length; j++)
+            {
+                var parameter = parameters[j];
+                if (parameter.ParameterType.IsGenericParameter && parameter.ParameterType.GenericParameterPosition == i)
+                {
+                    return args[i].Type;
+                }
+                if (parameter.ParameterType.IsArray)
+                {
+                    var elementType = parameter.ParameterType.GetElementType();
+                    var genericArgName = elementType.Name;
+                    var genericParameterPosition = elementType.GenericParameterPosition;
+                }
+                else if (parameter.ParameterType.IsGenericType)
+                {
+                    var value = GetGenericParameterType(genericArgument, parameter.ParameterType.GetGenericArguments(), args[j].Type);
+                    if (value is Type) return value;
+                }
+            }
+            return null;
+        }
+
+        private static Type GetGenericParameterType(Type genericArg, Type[] searchedGenericTypes, Type expressionType)
+        {
+            if (searchedGenericTypes is object && searchedGenericTypes.Length > 0)
```

Comment:
When can be `searchedGenericTypes` null or empty? This method is only called when the type is generic, so it should contain at least one type argument